src/main/java/com/stripe/net/HttpHeaders.java

```diff
@@ -0,0 +1,80 @@
+package com.stripe.net;
+
+import static java.util.Objects.requireNonNull;
+
+import com.stripe.util.CaseInsensitiveMap;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lombok.EqualsAndHashCode;
+
+/**
+ * A read-only view of a set of HTTP headers.
+ *
+ * <p>This class mimics the {@code java.net.http.HttpHeaders} added in Java 11.
+ */
+@EqualsAndHashCode
+public class HttpHeaders {
+  private CaseInsensitiveMap<List<String>> headerMap;
+
+  private HttpHeaders(CaseInsensitiveMap<List<String>> headerMap) {
+    this.headerMap = headerMap;
+  }
+
+  /**
+   * Returns an HTTP headers from the given map.
```

Comment:
Tiny nit, but I wonder if we should say "Returns an HTTPHeaders ..." (i.e. the name of the object). The grammar doesn't really make sense unless it's implied that this is the name of a class.

Reply:
Haha, I copied the docstring straight from Java 11's `HttpHeader's` [docs](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpHeaders.html#of(java.util.Map,java.util.function.BiPredicate)), but I agree the grammar doesn't make sense. I'll update it 👍

Reply:
lol! They probably generated it with some docstring tool that deforms camel case as that is The Java Way ;) Thanks for the change though.